src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -67,6 +67,40 @@ def _can_access(self, request, member, organization):
 
         return False
 
+    def _can_admin_team(self, request, organization, team_slug):
+        acting_member = OrganizationMember.objects.get(
+            organization=organization,
+            user__id=request.user.id,
+            user__is_active=True,
+        )
+
+        # This only applies to admins, otherwise, should defer to `_can_access()`
+        # and its role management logic
+        if acting_member.role != 'admin':
+            return False
+
+        try:
+            team = Team.objects.get(
+                organization=organization,
+                slug=team_slug,
+            )
+        except Team.DoesNotExist:
+            return False
+
+        try:
+            # if acting member is in an admin role and is on the team
+            # then they can manage members of that team
+            OrganizationMemberTeam.objects.get(
+                team=team,
+                organizationmember=acting_member,
+            )
+
+            return True
+        except OrganizationMemberTeam.DoesNotExist:
```

Comment:
You should be able to combine all three queries into a single one using joins/nested conditions.

```
organization=organization,
user__id=request.user.id,
user__is_active=True,
role='admin',
organizationmemberteam__team__slug==team_slug,
```

Should be close. I wasn't able to run this due to being on a phone.